css/components/badges.css

```diff
@@ -45,24 +49,89 @@
 }
 
 /* Avatar Badges positioning */
-.badge-wrapper .avatar-lg ~ .badge {
-  right: 0.75rem;
-  bottom: 2rem;
+/* top-left */
+.avatar-lg ~ .badge.top-left {
+  top: var(--scale-075);
+  left: var(--scale-075);
+}
+
+.avatar-md ~ .badge.top-left {
+  top: calc(var(--scale-05) - var(--size-px));
+  left: calc(var(--scale-05) - var(--size-px));
+}
+
+.avatar-sm ~ .badge.top-left {
+  top: calc(var(--size-3) / 2);
+  left: calc(var(--size-1) - 6px);
+}
+
+.avatar-xs ~ .badge.top-left {
+  top: calc(var(--size-1) - 6px);
+  left: calc(var(--size-1) - 8px);
+}
+
+/* top-right */
+
+.avatar-lg ~ .badge.top-right {
+  top: var(--scale-075);
+  right: var(--scale-075);
+}
+
+.avatar-md ~ .badge.top-right {
+  top: calc(var(--scale-05) - var(--size-px));
+  right: calc(var(--scale-05) - var(--size-px));
+}
+
+.avatar-sm ~ .badge.top-right {
+  top: calc(var(--size-3) / 2);
+  right: calc(var(--size-1) - 6px);
+}
+
+.avatar-xs ~ .badge.top-right {
+  top: calc(var(--size-1) - 6px);
+  right: calc(var(--size-1) - 8px);
+}
+
+/* bottom-right */
+
+.avatar-lg ~ .badge.bottom-right {
+  bottom: var(--scale-1);
+  right: var(--scale-075);
 }
 
-.badge-wrapper .avatar-md ~ .badge {
-  right: 0.8rem;
-  bottom: 1.2rem;
+.avatar-md ~ .badge.bottom-right {
+  bottom: var(--scale-075);
+  right: calc(var(--scale-05) - 2px);
 }
 
-.badge-wrapper .avatar-sm ~ .badge {
-  right: 6px;
-  bottom: 1.2rem;
+.avatar-sm ~ .badge.bottom-right {
+  bottom: calc(var(--size-5) / 2);
+  right: 0px;
 }
 
-.badge-wrapper .avatar-xs ~ .badge {
-  right: 6px;
-  bottom: 12px;
-  width: 0.75rem;
-  height: 0.75rem;
+.avatar-xs ~ .badge.bottom-right {
+  bottom: calc(var(--size-1) / 2);
+  right: calc(var(--size-1) - 8px);
```

Comment:
any specific purpose for not using --size-2 instead of `8px`?

Reply:
not specific but wanted to use only one variable here.